src/lxilua.c

```diff
@@ -254,6 +264,11 @@ static int scpi_raw(lua_State *L)
     const char *command = lua_tostring(L, 2);
     int timeout = lua_tointeger(L, 3);
 
+    if(device < 0)
+    {
+        lua_pushlstring(L, -1);
```

Comment:
I think this should be lua_pushnumber(L, -1); the same way it returns in the end of this function.